internal/controller/ipaddressclaim_controller.go

```diff
@@ -148,7 +148,7 @@ func (r *IpAddressClaimReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 		// 6.a create the IPAddress object
 		ipAddressResource := generateIpAddressFromIpAddressClaim(o, ipAddressModel.IpAddress)
-		err = controllerutil.SetOwnerReference(o, ipAddressResource, r.Scheme)
+		err = controllerutil.SetControllerReference(o, ipAddressResource, r.Scheme)
```

Comment:
Can you help me understand what's the rationale behind this change :)

Reply:
By default the `Owns()` function in the `SetupWithManager` function reconciles the first controller-type owner reference. With this change a controller-type owner reference is added to the ipAddressResource and `Owns()` will trigger the reconciliation of the ip address claim resource if a change in the ip address resource is detected.